pkg/fileutil/files.go

```diff
@@ -135,6 +135,26 @@ func IsDirectory(path string) bool {
 	return fileInfo.IsDir()
 }
 
+// IsDirectoryEmpty returns true if path is a dir and it's empty
+func IsDirectoryEmpty(path string) bool {
+	if IsDirectory(path) {
+		f, err := os.Open(path)
+		if err != nil {
+			return false
+		}
+		defer f.Close()
+
+		// read in ONLY one file
+		_, err = f.Readdir(1)
```

Comment:
Pretty sure `.` and `..` are always there. Does this work right?

Reply:
I took this code out of https://socketloop.com/tutorials/golang-determine-if-directory-is-empty-with-os-file-readdir-function - it did work on all my tests.

Reply:
OK, as long as you've debugged through it. Oh, please add a test. Most of the functions in there have tests, maybe all of them.

Reply:
Also, please include a link in the code comment when you've copied code from somewhere, thanks!

Reply:
Code added

Reply:
Test added!